test/hamiltonian/test_heat.py

```diff
@@ -183,6 +158,36 @@ def create_generator(
 
         return generator_ham
 
+    def test_transpile_options_sync(self):
+        """Test if transpile option set to composite can update all component experiments."""
+        exp = ZX90HeatXError(qubits=(0, 1), backend=self.backend)
+        basis_exp0 = exp.component_experiment(0).transpile_options.basis_gates
+        basis_exp1 = exp.component_experiment(1).transpile_options.basis_gates
+        self.assertListEqual(basis_exp0, ["sx", "x", "rz", "heat"])
+        self.assertListEqual(basis_exp1, ["sx", "x", "rz", "heat"])
```

Comment:
Minor, but why do you need to test the default values?

Reply:
This is pre-check to test if basis is really updated (from "heat" to "my_heat") synchronously.